internals/resource-managers/output.go

```diff
@@ -0,0 +1,177 @@
+	outputSpec := loggingv1beta1.OutputSpec{
+		SplunkHecOutput: &output.SplunkHecOutput{
+			HecHost:     logSpec.Host,
+			HecPort:     SplunkHecPort,
+			InsecureSSL: &insecureSSL,
+			HecToken: &secret.Secret{
+				ValueFrom: &secret.ValueFrom{
+					SecretKeyRef: &v1.SecretKeySelector{
+						LocalObjectReference: v1.LocalObjectReference{Name: logSpec.HecTokenSecretName},
+						Key:                  "SplunkHecToken",
+					},
+				},
+			},
+			Index:    logSpec.Index,
+			Protocol: protocol,
+			Format:   &output.Format{Type: "json"},
+			Buffer: &output.Buffer{
+				Timekey:       BufferTimekey,
+				TimekeyWait:   BufferTimekeyWait,
+				TimekeyUseUtc: BufferTimekeyUseUtc,
+			},
+		},
+	}
+	return outputSpec
+}
+
+// outputCreators is a map that associates log types with their corresponding output creation functions.
+var outputCreators = map[string]func(rcsv1alpha1.LogSpec) loggingv1beta1.OutputSpec{
+	LogTypeElastic: createElasticsearchOutput,
+	LogTypeSplunk:  createSplunkHecOutput,
+}
+
+// prepareResource prepares an output resource based on the provided capp.
+func (o OutputManager) prepareResource(capp rcsv1alpha1.Capp) loggingv1beta1.Output {
+	outputName := capp.GetName() + "-output"
+	if createFunc, ok := outputCreators[capp.Spec.LogSpec.Type]; ok {
+		outputSpec := createFunc(capp.Spec.LogSpec)
+		output := loggingv1beta1.Output{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      outputName,
+				Namespace: capp.GetNamespace(),
+			},
+			Spec: outputSpec,
+		}
+		return output
+	}
+	return loggingv1beta1.Output{}
+}
+
+// CleanUp deletes the output resource associated with the Capp object.
+// The output resource is deleted by calling the DeleteResource method of the resourceManager object.
+func (o OutputManager) CleanUp(capp rcsv1alpha1.Capp) error {
+	outputName := capp.GetName() + "-output"
+	resourceManager := rclient.ResourceBaseManager{Ctx: o.Ctx, K8sclient: o.K8sclient, Log: o.Log}
+	output := loggingv1beta1.Output{}
+	if err := resourceManager.DeleteResource(&output, outputName, capp.Namespace); err != nil {
+		return err
+	}
+	return nil
+}
+
+// CreateOrUpdateObject creates or updates an output object based on the provided capp.
+// It returns an error if any operation fails.
+func (o OutputManager) CreateOrUpdateObject(capp rcsv1alpha1.Capp) error {
+	outputName := capp.GetName() + "-output"
+	logger := log.FromContext(o.Ctx).WithValues("CappName", capp.Name, "CappNamespace", capp.Namespace, "outputName", outputName)
+	if capp.Spec.LogSpec.Type == LogTypeElastic || capp.Spec.LogSpec.Type == LogTypeSplunk {
+		generatedOutput := o.prepareResource(capp)
+		// get instance of current output
+		currentOutput := loggingv1beta1.Output{}
+		resourceManager := rclient.ResourceBaseManager{Ctx: o.Ctx, K8sclient: o.K8sclient, Log: o.Log}
+		logger.Info("trying to fetch existing output")
+		if err := o.K8sclient.Get(o.Ctx, types.NamespacedName{Namespace: capp.Namespace, Name: outputName}, &currentOutput); err != nil {
```

Comment:
instead of this if use a switch case when you take series of ifs which relate to the same var ill use a switch case its cleaner .